Changes to support Windows

fredvisser opened the pull request:
1. `subprocess.Popen()` has platform specific arguments. Added a platform check before calling a Windows compatible version of the call.
2. Replaced a manual path join with a `os.path.join` call (and removed unnecessary `/`)
3. Added `encoding='utf-8', errors='ignore'` parameters to the file.open call to resolve issues where some files contained un-parseable characters.

I've done basic testing to confirm that it works correctly on Windows and macOS.

fredvisser committed:
Changes to support Windows

fredvisser committed:
Minor simplification

initialed85 commented on third_party_license_file_generator/site_packages.py:
```diff
@@ -241,7 +255,7 @@ def _read_all_module_metadatas_and_license_files(self):
                 self._module_licenses_by_module_name[module_name] = license_file
 
     def _read_license(self, license_path):
```
This fails for me with Python2.7 (I know, I know, nobody should use it!)- I believe it can be worked around with the following import line:

`from codecs import open`

This will replace the open method with one that works in the desired way (for both Python2.7 and Python3.x)

initialed85 commented:
Thanks for the PR!

Looks good to me overall- I tested with that changed import line (see comment in `site_packages.py` and it all still works on MacOS.

Another request- I seem to have overlooked the `PyYAML` requirement in `setup.py` under the `install_requires` part; are you able to fix that up on my behalf?

initialed85 commented on third_party_license_file_generator/site_packages.py:
```diff
@@ -241,7 +255,7 @@ def _read_all_module_metadatas_and_license_files(self):
 
     def _read_license(self, license_path):
-        with open(license_path, 'r') as f:
+        with open(license_path, 'r', encoding='utf-8', errors='ignore') as f:
```
This fails for me with Python2.7 (I know, I know, nobody should use it!)- I believe it can be worked around with the following import line:

`from codecs import open`

This will replace the open method with one that works in the desired way (for both Python2.7 and Python3.x)

fredvisser committed:
Addressing PR feedback

fredvisser commented:
Great feedback! 
1. Fixed the PyYAML requirement (and bumped it to 3.13 to resolve an install issue I saw on 3.12).
2. I've added the new import statement (`from codecs import open`). 

One minor note from testing with 2.7 and 3.7 – with 2.7 the license summary looks like:

``` 
# 2.7
Apache-2.0
        u'python-dateutil' by u'Gustavo Niemeyer <gustavo@niemeyer.net>' (u'https://dateutil.readthedocs.io')
        u'requests' by u'Kenneth Reitz <me@kennethreitz.org>' (u'http://python-requests.org')
```

```
# 3.7
Apache-2.0
        'python-dateutil' by 'Gustavo Niemeyer <gustavo@niemeyer.net>' ('https://dateutil.readthedocs.io')
        'requests' by 'Kenneth Reitz <me@kennethreitz.org>' ('http://python-requests.org')
```

initialed85 commented:
> Great feedback!
> 
> 1. Fixed the PyYAML requirement (and bumped it to 3.13 to resolve an install issue I saw on 3.12).
> 2. I've added the new import statement (`from codecs import open`).
> 
> One minor note from testing with 2.7 and 3.7 – with 2.7 the license summary looks like:
> 
> ```
> # 2.7
> Apache-2.0
>         u'python-dateutil' by u'Gustavo Niemeyer <gustavo@niemeyer.net>' (u'https://dateutil.readthedocs.io')
>         u'requests' by u'Kenneth Reitz <me@kennethreitz.org>' (u'http://python-requests.org')
> ```
> 
> ```
> # 3.7
> Apache-2.0
>         'python-dateutil' by 'Gustavo Niemeyer <gustavo@niemeyer.net>' ('https://dateutil.readthedocs.io')
>         'requests' by 'Kenneth Reitz <me@kennethreitz.org>' ('http://python-requests.org')
> ```

Hmm- It's really just the user-facing summary with the main output being the generated licenses I guess.

I don't mind if you don't mind!

fredvisser commented:
> I don't mind if you don't mind!

I don't mind!

lowaa commented:
I've also tested it on MAC OS and it womm 👍 
Thanks for the update!
